processing/src/main/java/org/apache/druid/segment/filter/cnf/HiveCnfHelper.java

```diff
@@ -178,6 +179,11 @@ private static void generateAllCombinations(
           a.add(child);
           // Result must receive an actual OrFilter.
           result.add(idempotentOr(Filters.or(a)));
+          if (result.size() >= 10_000) {
```

Comment:
any reason for this limit to be configurable?

Reply:
this could be configurable - didn't keep it as of now to avoid more configs.
but if it feels like disruptive (because it could lead to failure in some existing queries which create > 10k filters and still run), I can add a config for this

Reply:
I've added the change where exceeding this limit for not fail queries for now. when the limit exceeds, the join flow will disable pushdown optimization and the normal filter flow will return the non-CNF filter. does that seem ok to not add a config to toggle the limit?

Reply:
yeah, i think that is ok, the transformation _shouldn't_ be changing results afaik, though per my other comment there should be some way to know that this has happened